src/Workspaces/Core/MSBuild.BuildHost/BuildHost.cs

```diff
@@ -161,9 +161,15 @@ private static ImmutableArray<(string ProjectPath, string ProjectGuid)> GetProje
     /// <summary>
     /// Returns the target ID of the <see cref="ProjectFile"/> object created for this.
     /// </summary>
-    public async Task<int> LoadProjectFileAsync(string projectFilePath, string languageName, CancellationToken cancellationToken)
+    public Task<int> LoadProjectFileAsync(string projectFilePath, string languageName, CancellationToken cancellationToken)
     {
         EnsureMSBuildLoaded(projectFilePath);
+        return LoadProjectFileCoreAsync(projectFilePath, languageName, cancellationToken);
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)] // Do not inline this, since this uses MSBuild types which are being loaded by the caller
```

Comment:
You may want to make a note that this was mono specific, lest somebody tries to undo this change being confused why it works on other platforms.